packages/commonwealth/client/scripts/views/pages/Community/Groups/common/GroupForm/GroupForm.tsx

```diff
@@ -0,0 +1,477 @@
+import React, { useEffect, useState } from 'react';
+import app from 'state';
+import { useFetchTopicsQuery } from 'state/api/topics';
+import { CWDivider } from 'views/components/component_kit/cw_divider';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWTextArea } from 'views/components/component_kit/cw_text_area';
+import { CWForm } from 'views/components/component_kit/new_designs/CWForm';
+import { CWSelectList } from 'views/components/component_kit/new_designs/CWSelectList';
+import { CWTextInput } from 'views/components/component_kit/new_designs/CWTextInput';
+import { MessageRow } from 'views/components/component_kit/new_designs/CWTextInput/MessageRow';
+import { CWButton } from 'views/components/component_kit/new_designs/cw_button';
+import { CWRadioButton } from 'views/components/component_kit/new_designs/cw_radio_button';
+import { ZodError } from 'zod';
+import TopicGatingHelpMessage from '../../TopicGatingHelpMessage';
+import './GroupForm.scss';
+import RequirementSubForm from './RequirementSubForm';
+import {
+  GroupFormProps,
+  RequirementSubType,
+  RequirementSubTypeWithLabel,
+} from './index.types';
+import {
+  VALIDATION_MESSAGES,
+  groupValidationSchema,
+  requirementSubFormValidationSchema,
+} from './validations';
+
+type CWRequirementsRadioButtonProps = {
+  inputError?: string;
+  inputValue: string;
+  onInputValueChange: (value: string) => any;
+};
+
+const CWRequirementsRadioButton = ({
```

Comment:
Ideally we could move `CWRequirementsRadioButton` into a separate file, this is the radio button with the text input in label.

I wouldn't recommend to move it to separate file, as its more meaningful here.

Reply:
this file has 500 lines of code and in this situation we really should move utils and other components to separate files. It does not cost us anything to create a new file with new component.